res/test_generate_update_metadata.py

```
@@ -0,0 +1,486 @@
+            "v\u0661.\u0662.\u0663",
+            "--version",
+            "\u0661.\u0662.\u0663",
+            check=False,
+        )
+
+        self.assertNotEqual(result.returncode, 0)
+    def test_release_id_validation_rejects_illegal_segments(self):
+        for bad_release_id in ["release/v1.4.6", "bad tag", "bad?tag", "bad#tag"]:
+            check = self.run_script("check-version", "--release-id", bad_release_id, "--version", "1.4.6", check=False)
+            self.assertNotEqual(check.returncode, 0, bad_release_id)
+            _, fragment = self.make_fragment(out_name=f"sign-source-{hashlib.sha1(bad_release_id.encode()).hexdigest()}.json")
```

Ruff (0.16.1)

[error] Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)